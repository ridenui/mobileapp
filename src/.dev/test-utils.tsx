// https://testing-library.com/docs/react-native-testing-library/setup
import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { light } from '@styles/Themes';

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={light}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

export * from '@testing-library/react-native';
export * from 'jest-styled-components';

export { customRender as render };
