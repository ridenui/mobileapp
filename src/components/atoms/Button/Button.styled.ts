import { ButtonSizes, ButtonVariants } from '@atoms/Button/Button';
import { Colors } from '@styles/Colors';
import type { Theme } from '@styles/Themes';
import styled from 'styled-components/native';

export type ButtonStyled = {
  variant: ButtonVariants;
  disabled: boolean;
  size: ButtonSizes;
};

type withTheme = ButtonStyled & { theme: Theme };

const getBackgroundColor = ({ variant, theme, disabled }: withTheme) => {
  if (variant === ButtonVariants.PRIMARY) {
    return {
      background: disabled ? theme['400'] : Colors.red,
    };
  }
  if (variant === ButtonVariants.SECONDARY) {
    return {
      background: disabled ? theme['400'] : Colors.orange,
    };
  }

  if (variant === ButtonVariants.GREEN) {
    return {
      background: disabled ? theme['400'] : Colors.green,
    };
  }

  return {};
};

const getSize = ({ size }: ButtonStyled) => {
  if (size === ButtonSizes.LARGE) {
    return {
      height: '32px',
    };
  }
  if (size === ButtonSizes.SMALL) {
    return {
      height: '24px',
    };
  }

  return {};
};

export const Button = styled.TouchableOpacity<ButtonStyled>`
  height: 32px;
  border-radius: 16px;
  justify-content: center;
  align-items: center;
  ${getBackgroundColor}
  ${getSize}
`;
