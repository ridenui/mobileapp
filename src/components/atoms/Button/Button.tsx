import React from 'react';
import type { TouchableOpacityProps, ViewStyle } from 'react-native';
import { Typography, TypographyVariants } from '@atoms/Typography/Typography';
import type { StyledProps } from 'styled-components';
import * as S from './Button.styled';

export enum ButtonVariants {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  FLAT = 'flat',
  GREEN = 'green',
}

export enum ButtonSizes {
  SMALL = 'small',
  LARGE = 'large',
}

export interface ButtonProps {
  /** Text Content of the Button. */
  children: string;
  /** onPress Handler */
  onPress?: TouchableOpacityProps['onPress'];
  /** Set button disabled state */
  disabled?: TouchableOpacityProps['disabled'];
  /** custom styles */
  style?: StyledProps<ViewStyle>;
  variant?: ButtonVariants;
  textStyle?: TypographyVariants;
  size?: ButtonSizes;
}

/**
 * It's a Button.
 */
export function Button({
  children,
  onPress,
  disabled,
  style,
  variant = ButtonVariants.PRIMARY,
  textStyle = TypographyVariants.Paragraph,
  size = ButtonSizes.LARGE,
}: ButtonProps): JSX.Element {
  return (
    <S.Button onPress={onPress} disabled={!!disabled} style={style} variant={variant} size={size}>
      <Typography variant={textStyle}>{children}</Typography>
    </S.Button>
  );
}
