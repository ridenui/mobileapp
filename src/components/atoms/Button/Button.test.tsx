import React from 'react';
import { render } from '@dev/test-utils';
import { Button } from './Button';

test('Button', () => {
  const { toJSON } = render(<Button>Button</Button>);
  expect(toJSON()).toMatchSnapshot();
});
