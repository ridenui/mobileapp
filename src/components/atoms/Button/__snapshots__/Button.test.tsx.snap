// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D1D1D6",
      "borderRadius": 16,
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
      ]
    }
  >
    Button
  </Text>
</View>
`;
